src/figuresClasses.ts

```diff
@@ -1,19 +1,83 @@
-export interface Figure {
+type Color = 'red' | 'green' | 'blue';
 
+enum FigureType {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
 }
 
-export class Triangle {
-
+export interface Figure {
+  color: string;
+  shape: FigureType;
+  getArea(): number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape = FigureType.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a < 1 || this.b < 1 || this.c < 1) {
+      throw new Error('Wrong value for Triangle');
+    }
+
+    if (this.a + this.b <= this.c) {
+      throw new Error('Wrong value for Triangle');
+    }
```

Comment:
Are you sure about this? What if we have a = 3, b = 1, c = 1 ? Check all cases :)